File: api/src/main/kotlin/api/db/models/Itinerary.kt
package api.db.models

data class Itinerary(
    val itineraryCode: String,
    val fullLineCode: String,
    val direction: Int,
    val tripId: String,
    val serviceId: String,
    val tripName: String,
)

data class ItineraryWithStops(
    val itineraryCode: String,
    val fullLineCode: String,
    val direction: Int,
    val stops: List<StopOrder>,
    val tripId: String?,
)

data class StopOrder(
    val fullStopCode: String,
    val tripId: String,
    val order: Int,
    val departureTime: Long,
)

data class StopOrderWithItineraries(
    val fullStopCode: String,
    val tripId: String,
    val itineraries: List<Itinerary>,
    val order: Int,
    val departureTime: Long,
)

File: api/Data/Entities/Itinerary.cs
namespace MadridTransporte.Api.Data.Entities;

public class Itinerary
{
    public int Id { get; set; }
    public required string ItineraryCode { get; set; }
    public required string FullLineCode { get; set; }
    public int Direction { get; set; }
    public required string TripId { get; set; }
    public required string ServiceId { get; set; }
    public required string TripName { get; set; }
}
